server/service/integration_ddm_test.go

```diff
@@ -0,0 +1,910 @@
+
+		r, err = mdmDevice.DeclarativeManagement("declaration-items")
+		require.NoError(t, err)
+		checkDeclarationItemsResp(t, parseDeclarationItemsResp(r), currDeclToken, mapDeclsByChecksum(noTeamDeclsByUUID))
+		checkRequestsDatabase(t, "declaration-items", mdmDevice.UUID, 2)
+	})
+
+	t.Run("Status", func(t *testing.T) {
+		checkRequestsDatabase(t, "status", mdmDevice.UUID, 0)
+		_, err := mdmDevice.DeclarativeManagement("status", fleet.MDMAppleDDMStatusReport{})
+		require.NoError(t, err)
+		checkRequestsDatabase(t, "status", mdmDevice.UUID, 1)
+	})
+
+	t.Run("Declaration", func(t *testing.T) {
+		want := noTeamDeclsByUUID["123"]
+		declarationPath := fmt.Sprintf("declaration/%s/%s", "configuration", want.Identifier)
+		checkRequestsDatabase(t, declarationPath, mdmDevice.UUID, 0)
+		r, err := mdmDevice.DeclarativeManagement(declarationPath)
+		require.NoError(t, err)
+
+		assertDeclarationResponse(r, want)
+
+		// insert a new declaration
+		noTeamDeclsByUUID["abc"] = fleet.MDMAppleDeclaration{
+			DeclarationUUID: "abc",
+			TeamID:          ptr.Uint(0),
+			Identifier:      "com.example4",
+			Name:            "Example4",
+			RawJSON: json.RawMessage(`{
+				"Type": "com.apple.configuration.test",
+				"Payload": {"foo":"bar"},
+				"Identifier": "com.example4"
+			}`),
+			CreatedAt:  then.Add(3 * time.Minute),
+			UploadedAt: then.Add(3 * time.Minute),
+		}
+		insertDeclaration(t, noTeamDeclsByUUID["abc"])
+		insertHostDeclaration(t, mdmDevice.UUID, noTeamDeclsByUUID["abc"])
+		want = noTeamDeclsByUUID["abc"]
+		r, err = mdmDevice.DeclarativeManagement(fmt.Sprintf("declaration/%s/%s", "configuration", want.Identifier))
+		require.NoError(t, err)
+		checkRequestsDatabase(t, declarationPath, mdmDevice.UUID, 1)
+
+		// try getting a non-existent declaration, should fail 404
+		nonExistantDeclarationPath := fmt.Sprintf("declaration/%s/%s", "configuration", "nonexistent")
+		checkRequestsDatabase(t, nonExistantDeclarationPath, mdmDevice.UUID, 0)
+		_, err = mdmDevice.DeclarativeManagement(nonExistantDeclarationPath)
+		require.Error(t, err)
+		require.ErrorContains(t, err, "404 Not Found")
+		checkRequestsDatabase(t, nonExistantDeclarationPath, mdmDevice.UUID, 1)
+
+		// typo should fail as bad request
+		typoDeclarationPath := fmt.Sprintf("declarations/%s/%s", "configurations", want.Identifier)
+		checkRequestsDatabase(t, typoDeclarationPath, mdmDevice.UUID, 0)
+		_, err = mdmDevice.DeclarativeManagement(typoDeclarationPath)
+		require.Error(t, err)
+		require.ErrorContains(t, err, "400 Bad Request")
+		checkRequestsDatabase(t, typoDeclarationPath, mdmDevice.UUID, 1)
+
+		assertDeclarationResponse(r, want)
+	})
+}
+
+func (s *integrationMDMTestSuite) TestAppleDDMReconciliation() {
+	t := s.T()
+	ctx := context.Background()
+	// TODO: use config logger or take into account FLEET_INTEGRATION_TESTS_DISABLE_LOG
+	logger := kitlog.NewJSONLogger(os.Stdout)
+
+	// TODO: use endpoints once those are available.
```

Comment:
I noticed a couple more TODOs like that in the rest of the tests, just pointing out in case the story is sufficiently advanced to use the API for those checks. Maybe they are meant to be addressed later.

Reply:
I really want to get to those during the freeze, I will ask George to spend the rest of the freeze just writing/improving tests and bugs for DDM